include/envoy/stats/symbol_table.h

```diff
@@ -183,6 +197,22 @@ class SymbolTable {
    *
    */
   virtual StoragePtr encode(absl::string_view name) PURE;
+
+  /**
+   * Enables trackRecentLookups to also track lookups that occur in
```

Comment:
Would it be better to have the stat name set be created from the symbol table such that a reference can be embedded and this handled automatically? Is it this way because of the extra memory that would require?

Reply:
> one option I guess is to have SymbolTable::makeSet(). That could also work, but would be an API change for all the existing StatNameSets. If you think that is a lot better we can change it though.

IMO that would be cleaner, yes, if you don't mind.

Reply:
don't mind; will do that a bit later. will be non-controversial but a few files hit.

Reply:
ok; done...this added a bunch of files to the PR and we could potentially break that out as a separate one if desired.

Reply:
resolving assuming the PR is not too big to review now.